### ISSUE TITLE
[tablets] nodetool refresh --load-and-stream loses data

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylladb#17906]
Seen in e.g. https://jenkins.scylladb.com/view/master/job/scylla-master/job/tablets/job/gating-dtest-release-with-tablets/1/testReport/migration_test/TestMigration/FullDtest___full_split000___test_migrate_sstable_with_wide_row_2_1_x_/
Scylla version 5.5.0~dev-0.20240319.e48eb76f6104
```
self = <migration_test.TestMigration object at 0x7f18b4921350>

    def test_migrate_sstable_with_wide_row(self):
        node1 = self.start_cluster_and_get_node1()
    
        # CREATE COLUMNFAMILY ks.cf (key varchar, c varchar, v varchar, PRIMARY KEY(key, c))
        self.create_ks_and_cf(node1, None, None, False)
        self.load_migrated_tables(node1, "with_wide_rows")
    
>       self.check_number_of_rows(node1, 3)

migration_test.py:312: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <migration_test.TestMigration object at 0x7f18b4921350>
node = <ccmlib.scylla_node.ScyllaNode object at 0x7f18b2af1fd0>
expected_number_of_rows = 3

    def check_number_of_rows(self, node, expected_number_of_rows):
        logger.info("Checking rows on node1...")
        query = "SELECT COUNT(*) FROM cf"
        statement = SimpleStatement(query)
        s = self.patient_cql_connection(node, "ks")
        result = list(s.execute(statement))
>       assert result[0].count == expected_number_of_rows, "Expected {} rows. Got {}".format(expected_number_of_rows, list(s.execute("SELECT * FROM ks.cf")))
E       AssertionError: Expected 3 rows. Got [Row(key='a', c='a', v='b')]
E       assert 1 == 3
E        +  where 1 = Row(count=1).count

migration_test.py:698: AssertionError
```

`load_migrated_tables` executes nodetool refresh in the uploads directory, and since tablets are enabled we implicitly enable load-and-stream.

https://jenkins.scylladb.com/job/scylla-master/job/tablets/job/gating-dtest-release-with-tablets/1/artifact/logs-full.release.000/1710885604404_migration_test.py%3A%3ATestMigration%3A%3Atest_migrate_sstable_with_wide_row%5B2_1_x%5D/node1.log
```
INFO  2024-03-19 22:00:01,129 [shard 1:strm] sstables_loader - Loading new SSTables for keyspace=ks, table=cf, load_and_stream=auto-enabled-for-tablets, primary_replica_only=false
INFO  2024-03-19 22:00:01,132 [shard 1:strm] sstables_loader - load_and_stream: started ops_uuid=246eebf8-504f-4b47-b569-b0b7c9ea63f1, process [0-1] out of 1 sstables={/jenkins/workspace/scylla-master/tablets/gating-dtest-release-with-tablets/scylla/.dtest/dtest-t715_iwn/test/node1/data/ks/cf-07691ee0e63c11ee975fa8ba6d78ae51/upload/ks-cf-ka-1-Data.db}
INFO  2024-03-19 22:00:01,137 [shard 1:strm] stream_session - [Stream #246eebf8-504f-4b47-b569-b0b7c9ea63f1] Write to sstable for ks=ks, cf=cf, estimated_partitions=256, received_partitions=1
INFO  2024-03-19 22:00:01,137 [shard 1:strm] sstables_loader - load_and_stream: ops_uuid=246eebf8-504f-4b47-b569-b0b7c9ea63f1, ks=ks, table=cf, target_node=127.0.89.1, num_partitions_sent=1, num_bytes_sent=207
INFO  2024-03-19 22:00:01,137 [shard 1:strm] sstables_loader - load_and_stream: finished ops_uuid=246eebf8-504f-4b47-b569-b0b7c9ea63f1, ks=ks, table=cf, partitions_processed=1 partitions, bytes_processed=207 bytes, partitions_per_second=183.0969 partitions/s, bytes_per_second=0.036145266 MiB/s, duration=0.005461589 s, status=succeeded
INFO  2024-03-19 22:00:01,138 [shard 1:strm] sstables_loader - Done loading new SSTables for keyspace=ks, table=cf, load_and_stream=true, primary_replica_only=false, status=succeeded
```

we see above `partitions_processed=1` while the sstable should have 3 partitions

**Comment by bhalevy** [comment]
@asias please look into this.  Is it a known issue?
Cc @denesb

**Comment by denesb** [comment]
Maybe fixed by this: https://github.com/scylladb/scylladb/pull/17899
@raphaelsc

**Comment by bhalevy** [comment]
This is seen in several dtests, for example:
https://jenkins.scylladb.com/view/master/job/scylla-master/job/tablets/job/gating-dtest-release-with-tablets/1/testReport/migration_test/TestMigration/FullDtest___full_split000___test_migrate_sstable_with_frozen_collection_map_3_0_md_/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/tablets/job/gating-dtest-release-with-tablets/1/testReport/migration_test/TestLoadAndStream/FullDtest___full_split000___test_load_and_stream_decrease_cluster_3_0_md_/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/tablets/job/gating-dtest-release-with-tablets/1/testReport/migration_test/TestLoadAndStream/FullDtest___full_split000___test_load_and_stream_frozen_pk_3_0_md_/
and possibly more

**Comment by denesb** [comment]
Should be fixed by https://github.com/scylladb/scylladb/pull/17899.
Closing tentatively.

**Comment by raphaelsc** [comment]
> Should be fixed by #17899. Closing tentatively.

https://jenkins.scylladb.com/view/master/job/scylla-master/job/tablets/job/gating-dtest-release-with-tablets/5/#showFailuresLink confirms it's fixed

**Comment by bhalevy** [comment]
> > Should be fixed by #17899. Closing tentatively.
> 
> https://jenkins.scylladb.com/view/master/job/scylla-master/job/tablets/job/gating-dtest-release-with-tablets/5/#showFailuresLink confirms it's fixed

https://jenkins.scylladb.com/view/master/job/scylla-master/job/tablets/job/gating-dtest-release-with-tablets/5/testReport/migration_test/TestLoadAndStream/ in particular

**Comment by raphaelsc** [comment]
> > > Should be fixed by #17899. Closing tentatively.
> > 
> > 
> > https://jenkins.scylladb.com/view/master/job/scylla-master/job/tablets/job/gating-dtest-release-with-tablets/5/#showFailuresLink confirms it's fixed
> 
> https://jenkins.scylladb.com/view/master/job/scylla-master/job/tablets/job/gating-dtest-release-with-tablets/5/testReport/migration_test/TestLoadAndStream/ in particular

indeed, thanks.
